Moblin/Various/Model.swift

```diff
@@ -1204,13 +1205,14 @@ final class Model: NSObject, ObservableObject {
         DispatchQueue.main.async {
             let camera = self.getRtmpStream(streamKey: streamKey)?.camera() ?? rtmpCamera(name: "Unknown")
             self.makeToast(title: "\(camera) disconnected")
-            guard let cameraId = self.getRtmpStream(streamKey: streamKey)?.id else {
+            guard let stream = self.getRtmpStream(streamKey: streamKey) else {
                 return
             }
-            self.media.removeRtmpCamera(cameraId: cameraId)
+            self.media.removeRtmpCamera(cameraId: stream.id)
             if self.database.debug!.enableRtmpAudio! {
-                self.media.removeRtmpAudio(cameraId: cameraId)
+                self.media.removeRtmpAudio(cameraId: stream.id)
             }
+            self.selectMic(mic: self.previousMic)
```

Comment:
What happens if the previous mic also was an RTMP mic, and that mic can no longer be used as that RTMP stream was disconnected some time ago?

Reply:
Or if it was an bluetooth mic. The same logic can be applied to those I think.

Reply:
It might be easiest to just start using a builtin mic if an external or RTMP mic is disconnected.

Reply:
I suggest I switch to the microphone, which is saved in the settings. that would be front, bottom or back

Reply:
sounds good. didn't remember that exists =)

Reply:
or top of course :)